cpp/src/groupby/hash/groupby.cu

```diff
@@ -667,10 +654,18 @@ std::pair<std::unique_ptr<table>, std::vector<aggregation_result>> groupby(
   rmm::cuda_stream_view stream,
   rmm::mr::device_memory_resource* mr)
 {
+  auto const has_list_column =
+    std::any_of(keys.begin(), keys.end(), [](cudf::column_view const& col) {
+      return col.type().id() == type_id::LIST;
+    });
+  if (has_list_column and include_null_keys == cudf::null_policy::EXCLUDE) {
+    CUDF_FAIL("Null LIST keys cannot be excluded.");
+  }
```

Comment:
Should this be applied to struct as well? If so, just check `is_nested(data_type)`.